src/renderer/components/deposit/withdraw/Withdraw.tsx

```diff
@@ -1,284 +1,339 @@
-      )
+    const runeBalance = FP.pipe(
+      oRuneBalance,
+      O.getOrElse(() => ZERO_BASE_AMOUNT)
     )
-  }, [oFees, withdrawAmounts.runeWithdraw, withdrawPercent])
-
-  const renderThorOutFeeError = useMemo(() => {
-    if (!isThorOutFeeError) return <></>
 
     return FP.pipe(
-      oFees,
-      O.map(({ thorOut }) => {
+      oFee,
+      O.map((fee) => {
         const msg = intl.formatMessage(
-          { id: 'deposit.withdraw.add.error.outFeeNotCovered' },
+          { id: 'deposit.withdraw.error.feeNotCovered' },
           {
             fee: formatAssetAmountCurrency({
-              amount: baseToAsset(thorOut),
+              amount: baseToAsset(fee),
               asset: AssetRuneNative,
               trimZeros: true
             }),
-            amount: formatAssetAmountCurrency({
-              amount: withdrawAmounts.runeWithdraw,
+            balance: formatAssetAmountCurrency({
+              amount: baseToAsset(runeBalance),
               asset: AssetRuneNative,
               trimZeros: true
             })
           }
         )
-        // `key`  has to be set for any reason to avoid "Missing "key" prop for element in iterator"
-        return <Styled.FeeErrorLabel key="thor-fee-out-error">{msg}</Styled.FeeErrorLabel>
+        return <Styled.FeeErrorLabel key="fee-error">{msg}</Styled.FeeErrorLabel>
       }),
       O.getOrElse(() => <></>)
     )
-  }, [isThorOutFeeError, oFees, intl, withdrawAmounts.runeWithdraw])
+  }, [isFeeError, oRuneBalance, oFee, intl])
+
+  // Deposit start time
+  const [depositStartTime, setDepositStartTime] = useState<number>(0)
+
+  // Withdraw state
+  const [withdrawState, setWithdrawState] = useState<WithdrawState>(INITIAL_WITHDRAW_STATE)
+
+  const txModalExtraContent = useMemo(() => {
+    const stepDescriptions = [
+      intl.formatMessage({ id: 'common.tx.healthCheck' }),
+      intl.formatMessage({ id: 'common.tx.sendingAsset' }, { assetSymbol: AssetRuneNative.symbol }),
+      intl.formatMessage({ id: 'common.tx.checkResult' })
+    ]
+    const stepDescription = FP.pipe(
+      withdrawState.withdraw,
+      RD.fold(
+        () => '',
+        () =>
+          `${intl.formatMessage(
+            { id: 'common.step' },
+            { current: withdrawState.step, total: withdrawState.stepsTotal }
+          )}: ${stepDescriptions[withdrawState.step - 1]}`,
+        () => '',
+        () => `${intl.formatMessage({ id: 'common.done' })}!`
+      )
+    )
 
-  const renderFee = useMemo(() => {
-    const loading = <>...</>
-    return FP.pipe(
-      fees,
+    return (
+      <DepositAssets
+        target={{ asset, amount: assetAmountToWithdraw }}
+        source={O.some({ asset: AssetRuneNative, amount: runeAmountToWithdraw })}
+        stepDescription={stepDescription}
+      />
+    )
+  }, [intl, asset, withdrawState, assetAmountToWithdraw, runeAmountToWithdraw])
+
+  const onCloseTxModal = useCallback(() => {
+    // unsubscribe
+    unsubscribeWithdrawSub()
```

Comment:
but `setWithdrawSub` already does it. is not it ?

Reply:
Correct, I will remove that line ^